### PR TITLE
RavenDB-22637 : XML documentation of IOperation

### PR DESCRIPTION
**Pull request by aviv86**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22637/XML-documentation-of-IOperation

### Additional description

- `Raven.Client.Documents.Operations.TimeSeries.GetMultipleTimeSeriesOperation`
- `Raven.Client.Documents.Operations.TimeSeries.GetTimeSeriesOperation`
- `Raven.Client.Documents.Operations.TimeSeries.GetTimeSeriesOperation`1`
- `Raven.Client.Documents.Operations.TimeSeries.GetTimeSeriesStatisticsOperation`
- `Raven.Client.Documents.Operations.TimeSeries.TimeSeriesBatchOperation`
- `Raven.Client.Documents.Operations.Counters.CounterBatchOperation`
- `Raven.Client.Documents.Operations.Counters.GetCountersOperation`
- `IRevisionIncludeBuilder<T, out TBuilder>`
- `ICompareExchangeValueIncludeBuilder<T, out TBuilder>`
- `IIncludeBuilder<T, out TBuilder>`

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs [hunk 1]
```diff
@@ -82,6 +88,9 @@ public enum CounterOperationType
         GetAll
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs:144]: let's put a note about each type here.
I wonder 
- what is `put` vs `increment` (would it override the value)
- `get` vs `getall` (if we specify a counter name)

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 2]
```diff
@@ -338,6 +345,9 @@ public DynamicJsonValue ToJson()
             }
         }
 
+        /// <summary>
+        /// Represents a delete operation in a time series, allowing data points within a specific range to be removed.
+        /// </summary>
         public sealed class DeleteOperation
         {
             public DateTime? From, To;
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: need to write something here as well
does `From` & `To` included in the deletion?

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 3]
```diff
@@ -291,6 +295,9 @@ public DynamicJsonValue ToJson()
             };
         }
 
+        /// <summary>
+        /// Represents an append operation in a time series, allowing new data points to be added at specific timestamps.
+        /// </summary>
         public sealed class AppendOperation
         {
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:335]: also needed
```
 public DateTime Timestamp;
 public double[] Values;
 public string Tag;
```

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 4]
```diff
@@ -364,6 +374,9 @@ public DynamicJsonValue ToJson()
             }
         }
 
+        /// <summary>
+        /// Represents an increment operation in a time series, allowing values at a specific timestamp to be incremented.
+        /// </summary>
         public sealed class IncrementOperation
         {
             public DateTime Timestamp;
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: same

File: src/Raven.Client/Documents/Session/TimeSeries/TimeSeriesEntry.cs [hunk 5]
```diff
@@ -18,6 +18,14 @@
...
+    /// Represents an individual entry in a time series, storing a timestamp, associated values, and metadata.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="TimeSeriesEntry"/> is used as a base type for time series data projections 
+    /// in operations such as <see cref="GetTimeSeriesOperation{TValues}"/>.
+    /// It contains a timestamp and associated values for a single entry within a time series.
+    /// </remarks>
     public class TimeSeriesEntry : ITimeSeriesQueryStreamEntry
     {
         public DateTime Timestamp { get; set; }
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Session/TimeSeries/TimeSeriesEntry.cs:35]: everything here has to be documented as well

**Review by karmeli87 — COMMENTED** [review]
For completeness sake let's also document the results and friends
- CountersDetail
- TimeSeriesDetails
- TimeSeriesRange

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs [hunk 6]
```diff
@@ -7,13 +7,19 @@
...
 namespace Raven.Client.Documents.Operations.Counters
 {
+    /// <summary>
+    /// Represents a batch of counter operations on multiple documents, supporting retrieval of counter values across nodes.
+    /// </summary>
     public sealed class CounterBatch
     {
         public bool ReplyWithAllNodesValues;
         public List<DocumentCountersOperation> Documents = new List<DocumentCountersOperation>();
         public bool FromEtl;
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: what about those?

File: src/Raven.Client/Documents/Operations/TimeSeries/GetTimeSeriesOperation.cs [hunk 7]
```diff
@@ -14,11 +14,17 @@ namespace Raven.Client.Documents.Operations.TimeSeries
 {
     public sealed class GetTimeSeriesOperation : GetTimeSeriesOperation<TimeSeriesEntry>
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GetTimeSeriesOperation"/> class, 
+        /// retrieving multiple entries from a time series associated with a document within a specified date range.
+        /// </summary>
+        /// <inheritdoc select="param" />
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/GetTimeSeriesOperation.cs:21]: I assume it works :)

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs [hunk 8]
```diff
@@ -8,6 +8,10 @@
 
 namespace Raven.Client.Documents.Operations.TimeSeries
 {
+    /// <summary>
+    /// Represents a batch operation for time series data, including appends, increments, and deletions, 
+    /// to be performed on a single document’s time series.
+    /// </summary>
     public sealed class TimeSeriesOperation
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:15]: let's document the other `public` here.
BTW does the `Name` is mandatory here?

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs [hunk 9]
```diff
@@ -7,13 +7,19 @@
...
     }
 
+    /// <summary>
+    /// Represents counter operations for a specific document, such as increment or set operations on named counters.
+    /// </summary>
     public sealed class DocumentCountersOperation
     {
         public List<CounterOperation> Operations;
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: What about other `public`s here?
does `DocumentId` is mandatory?

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs (hunk 1 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs:144]: done

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 2 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: done

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 3 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:335]: done

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 4 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs]: done

File: src/Raven.Client/Documents/Session/TimeSeries/TimeSeriesEntry.cs (hunk 5 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Session/TimeSeries/TimeSeriesEntry.cs:35]: done

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs (hunk 6 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: done

File: src/Raven.Client/Documents/Operations/TimeSeries/GetTimeSeriesOperation.cs (hunk 7 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/TimeSeries/GetTimeSeriesOperation.cs:21]: it does :)

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs (hunk 8 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesOperation.cs:15]: done
as yes, `Name` is mandatory - mentioned that in the docs

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs (hunk 9 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: done

**Review by karmeli87 — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**